src/model.py

```diff
@@ -0,0 +1,145 @@
+import helpers
+import classifiers
+import pandas as pd
+import matplotlib.pyplot as plt
+from sklearn.metrics import classification_report
+from sklearn.metrics import confusion_matrix
+from sklearn.decomposition import RandomizedPCA
+from config import *
+import os.path
+import pickle
+
+# Interface between the server/local helper and the neural network
+class Model(object):
+    def __init__(self, preprocessor, clf_id=MLP_ID, db=None):
+        self.preprocessor = preprocessor
+        self.clf_id = clf_id
+        self.clf = classifiers.choose_clf(clf_id)
+        self.db = db
+
+    def initialize(self):
+        cache_path = self.clf.cache_string()
+        if LOAD_CACHE and os.path.isfile(cache_path):
+            print("Loading model from %s" % cache_path)
+            with open(cache_path, 'rb') as f:
+                self.clf = pickle.load(f)
+        else:
+            print("Initializing model")
+            if self.db:
+                print("Loading data from DB")
+                data = self.preprocessor.data_from_db()
+            else:
+                print("Loading data from sklearn")
+                data = self.preprocessor.get_data()
+
+            (X_train, 
+            X_test, \
+            y_train, \
+            y_test, \
+            target_names, \
+            H, \
+            W
+            ) = data
+
+            pca = RandomizedPCA(n_components=PCA_N_COMPONENTS, whiten=True).fit(X_train)
+            eigenfaces = pca.components_.reshape((PCA_N_COMPONENTS, H, W))
+            X_train_processed = pca.transform(X_train)
+            self.train(X_train_processed, y_train)
+            self.save_model()
```

Comment:
L34-L48 is using the data to train a fresh model, whether the datas from db or the lfw database[1]. This only occurs if it couldn't find a cached already trained model. I'm assuming both are loaded in the same format. This is just photos and labels that have been divided for training and testing. (Testing not even used here actually)
Storing a cached model in DB is also possible and we should do that. Then it would check db and local for cache, if not load data to train.